cluster/addons/fluentd-elasticsearch/es-image/Dockerfile

```diff
@@ -11,44 +11,46 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
+FROM docker.elastic.co/elasticsearch/elasticsearch:5.5.0
 
-# A Dockerfile for creating an Elasticsearch instance that is designed
-# to work with Kubernetes logging. Inspired by the Dockerfile
-# dockerfile/elasticsearch
+USER root
 
-FROM java:openjdk-8-jre-alpine
-
-
-ENV DEBIAN_FRONTEND noninteractive
-ENV ELASTICSEARCH_VERSION 5.4.0
-
-RUN apk update && \
-    apk --no-cache add \
-        --repository https://dl-3.alpinelinux.org/alpine/edge/testing \
-        --repository https://dl-3.alpinelinux.org/alpine/edge/community \
-        curl \
-        shadow \
-        tar \
-        gosu
-
-RUN set -x \
-    && cd / \
-    && mkdir /elasticsearch \
-    && curl -O https://artifacts.elastic.co/downloads/elasticsearch/elasticsearch-$ELASTICSEARCH_VERSION.tar.gz \
-    && tar xf elasticsearch-$ELASTICSEARCH_VERSION.tar.gz -C /elasticsearch --strip-components=1 \
-    && rm elasticsearch-$ELASTICSEARCH_VERSION.tar.gz
-
-COPY config /elasticsearch/config
-
-COPY run.sh /
-COPY elasticsearch_logging_discovery /
-
-RUN useradd --no-create-home --user-group elasticsearch \
-    && mkdir /data \
-    && chown -R elasticsearch:elasticsearch /elasticsearch
+RUN mkdir /data
+RUN chown -R elasticsearch:elasticsearch /data
 
+WORKDIR /usr/share/elasticsearch
 
 VOLUME ["/data"]
 EXPOSE 9200 9300
 
-CMD ["/run.sh"]
+ENV GOSU_VERSION 1.10
+RUN set -ex; \
+  yum -y install epel-release; \
+  yum -y install wget dpkg; \
+  dpkgArch="$(dpkg --print-architecture | awk -F- '{ print $NF }')"; \
+  wget -O /usr/bin/gosu "https://github.com/tianon/gosu/releases/download/$GOSU_VERSION/gosu-$dpkgArch"; \
+  wget -O /tmp/gosu.asc "https://github.com/tianon/gosu/releases/download/$GOSU_VERSION/gosu-$dpkgArch.asc"; \
+  # verify the signature
+  export GNUPGHOME="$(mktemp -d)"; \
+  gpg --keyserver ha.pool.sks-keyservers.net --recv-keys B42F6819007F00F88E364FD4036A9C25BF357DD4; \
+  gpg --batch --verify /tmp/gosu.asc /usr/bin/gosu; \
+  rm -r "$GNUPGHOME" /tmp/gosu.asc; \
+  chmod +x /usr/bin/gosu; \
+  # verify that the binary works
+  gosu nobody true; \
+  yum -y remove wget dpkg; \
+  yum clean all
+
+USER elasticsearch
+# Temporarily move config out of the way before installing plugin as installation will fail
+# otherwise
+RUN mv config config.bak && ./bin/elasticsearch-plugin install -b repository-s3 && \
```

Comment:
if you have seen it I would love to get that in here.  there are several things that should be parameterized.